Run rustfmt on selectors, servo_arc, and style

bholley opened the pull request:
Now that rustfmt is getting close to stable, and work on the style system has died down a bit, it seemed like an opportune time to auto-format the style crates.

The first commit disables import reordering, since tidy and rustfmt don't currently agree on the correct ordering. The second commit does a bunch of manual fixups such that the output of rustfmt passes tidy. The third commit runs rustfmt on the three aforementioned crate.

There are a few dozen warnings in the style crate about lines longer than 100 characters. It would be good to fix these, but I don't have time for that now.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/20603)
<!-- Reviewable:end -->

bholley committed:
Manual fixups so that the rustfmt output won't trigger tidy.

bholley committed:
Disable import reordering for now.

Tidy wants it one way, and the latest nightly of rustfmt wants it
another way.

highfive commented:
Heads up! This PR modifies the following files:
 * @canaltinova: components/style/values/computed/pointing.rs, components/style/values/generics/flex.rs, components/style/stylesheets/style_rule.rs, components/style/gecko_bindings/sugar/ns_css_value.rs, components/style/invalidation/element/invalidator.rs and 166 more
 * @emilio: components/style/values/computed/pointing.rs, components/style/values/generics/flex.rs, components/style/stylesheets/style_rule.rs, components/style/gecko_bindings/sugar/ns_css_value.rs, components/style/invalidation/element/invalidator.rs and 166 more

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!
* These commits modify style code, but no tests are modified. Please consider adding a test!

bholley commented:
CC @upsuper @emilio @SimonSapin @pcwalton @nox @nrc

bholley commented:
(and @heycam of course)

r? @SimonSapin

bholley commented:
CC @Manishearth also

upsuper commented:
I wonder is this really just running rustfmt on the crates? It seems to be doing more things than I would thought a formatter normally does, e.g. removing `use` for those only used once.

Also I find it a bit hard to follow how some decision is made for formatting, for example, I see conversion from single line `if ... { A } else { B }` to multi-line, and vice versa.

upsuper commented:
Hmmm, actually not. Many of the changes I wouldn't expect formatter to do is from the manual fixup. That explains :)

emilio commented on components/selectors/parser.rs:
```diff
@@ -1901,33 +1904,32 @@ where
-                                )
-                            })?
-                        )
+                        let r = input.parse_nested_block(|input| {
```
what does `r` mean? Maybe just `selector`?

emilio commented:
CI is also red:

```
error: unused import: `malloc_size_of::MallocUnconditionalShallowSizeOf`
  --> components/style/stylesheets/style_rule.rs:10:5
   |
10 | use malloc_size_of::MallocUnconditionalShallowSizeOf;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
```

bors-servo commented:
:umbrella: The latest upstream changes (presumably #20606) made this pull request unmergeable. Please resolve the merge conflicts.

SimonSapin commented:
@bholley Could you include in the PR message (which will go in the merge commit) what version of rustfmt you used and what command you ran to produce this commit?

bholley commented:
Addressed review comments. If someone could stamp this soon I'd appreciate it, since it bitrots quickly. :-)

bors-servo commented:
:umbrella: The latest upstream changes (presumably #20613) made this pull request unmergeable. Please resolve the merge conflicts.

emilio commented on components/selectors/matching.rs:
```diff
@@ -267,12 +260,12 @@ where
+                matches!(
+                    selector.combinator_at_parse_order(from_offset),
+                    Combinator::SlotAssignment | Combinator::PseudoElement
+                )),
```
Don't understand how this makes more sense than what we had either but.. I guess.

emilio commented on components/selectors/matching.rs:
```diff
@@ -378,36 +364,33 @@ fn matches_hover_and_active_quirk<Impl: SelectorImpl>(
-            },
-            _ => true,
-        }
+    let all_match = selector_iter.clone().all(|simple| match *simple {
```
Ugh, this one looks really bad :(

emilio commented on components/selectors/matching.rs:
```diff
@@ -772,105 +737,80 @@ where
+            context
+                .shared
+                .shadow_host()
+                .map_or(false, |host| host == element.opaque()) &&
```
Ugh, wtf. This were two conditions before and now looks a bit messy...

emilio commented on components/selectors/parser.rs:
```diff
@@ -727,11 +756,11 @@ impl Combinator {
     /// Returns true if this combinator is a child or descendant combinator.
     #[inline]
     pub fn is_ancestor(&self) -> bool {
```
This function looks also pretty bad...

emilio commented on components/style/author_styles.rs:
```diff
@@ -57,27 +56,20 @@ where
-            flusher.sheets,
-            guard,
-        );
+        let _ = self.data
```
This looks like a regression too.

emilio commented on components/style/author_styles.rs:
```diff
@@ -57,27 +56,20 @@ where
-            /* host = */ None,
-            /* snapshot_map = */ None,
-        );
+        let flusher = self.stylesheets
```
This looks worse imo.

emilio commented on components/style/counter_style/mod.rs:
```diff
@@ -149,7 +156,7 @@ impl<'a, 'b, 'i> AtRuleParser<'i> for CounterStyleRuleParser<'a, 'b> {
 macro_rules! checker {
     ($self:ident._($value:ident)) => {};
-    ($self:ident.$checker:ident($value:ident)) => {
+    ($self:ident. $checker:ident($value:ident)) => {
```
This is touching a macro, and I'm not sure it's fine, actually... I guess it is but it looks fishy at best.

emilio commented on components/style/dom.rs:
```diff
@@ -338,15 +345,8 @@ pub trait TShadowRoot : Sized + Copy + Clone + PartialEq {
-    + Clone
-    + SelectorsElement<Impl = SelectorImpl>
+pub trait TElement:
+    Eq + PartialEq + Debug + Hash + Sized + Copy + Clone + SelectorsElement<Impl = SelectorImpl>
```
This looks also like a regression to me, but...

emilio commented on components/style/error_reporting.rs:
```diff
@@ -96,88 +104,107 @@ impl<'a> fmt::Display for ContextualParseError<'a> {
-                write!(f, "Invalid @counter-style rule: 'system: extends …' with 'additive-symbols'")
-            }
+            },
+            ContextualParseError::InvalidCounterStyleWithoutSymbols(ref system) => write!(
```
These look way worse too.

emilio commented on components/style/values/specified/outline.rs:
```diff
@@ -10,8 +10,8 @@ use selectors::parser::SelectorParseErrorKind;
-#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, Ord)]
-#[derive(PartialEq, PartialOrd, ToComputedValue, ToCss)]
+#[derive(Clone, Copy, Debug, Eq, MallocSizeOf, Ord, PartialEq, PartialOrd, ToComputedValue,
+         ToCss)]
```
It also merges `#[derive(..)]` lines? Is there a way to prevent it from doing that?

Manishearth reviewed (approved):
r=me

emilio commented on components/selectors/builder.rs:
```diff
@@ -266,28 +262,28 @@ impl From<u32> for Specificity {
-    )
-    where
-        Impl: SelectorImpl
+    ) where
```
This (`) where`) looks weird, is it expected?

bholley commented:
@bors-servo r=Manishearth

bors-servo commented:
:pushpin: Commit b3d39c4 has been approved by `Manishearth`

<!-- @bors-servo r=Manishearth b3d39c404e06f45310aa39357673c71bf131eed6 -->

emilio reviewed (approved):
There are a few of these that look like regressions to me. Maybe @nrc may take a look and see if there's something unexpected in some of these?

Also, is there any plan / way to enforce its usage? My plan to start doing this, btw, was enforcing usage of `rustfmt-format-diff` (https://github.com/rust-lang-nursery/rustfmt/blob/master/src/format-diff/main.rs) on incoming commits. But not sure if there's any better idea around.

Anyway, as much as I dislike some particular changes, consistency is king I guess...

bors-servo commented:
:lock: Merge conflict

bors-servo commented:
:umbrella: The latest upstream changes (presumably #20616) made this pull request unmergeable. Please resolve the merge conflicts.

bholley committed:
Run rustfmt on selectors, servo_arc, and style.

This was generated with:

./mach cargo fmt --package selectors &&
./mach cargo fmt --package servo_arc &&
./mach cargo fmt --package style

Using rustfmt 0.4.1-nightly (a4462d1 2018-03-26)

emilio commented:
FWIW I think something like #20617 would be less invasive. Arguably needs more polish and what not. But massive reformats pollute the history quite heavily, it's better to do it incrementally if possible IMO.

bholley commented:
The downsides I see to per-patch reformatting are:
(1) Patches that should otherwise be small (i.e. renames) can suddenly get big as they reformat surrounding code, making them harder to review.
(2) It's harder for new contributors to infer the correct style from surrounding code, because it may or may not be consistent with the canonical style.

There's definitely room for disagreement here. But given that most people have expressed approval for doing it this way, and given that I've already done the work here, I'm inclined to land it.

bholley commented:
@bors-servo r=Manishearth p=5

bors-servo commented:
:pushpin: Commit c99bcdd has been approved by `Manishearth`

<!-- @bors-servo r=Manishearth c99bcdd4b8d4b4ac495d8a26eb8a4d2c710a2589 -->

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/11275), [arm32](http://build.servo.org/builders/arm32/builds/11510), [arm64](http://build.servo.org/builders/arm64/builds/11549), [linux-dev](http://build.servo.org/builders/linux-dev/builds/11587), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/8476), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/8485), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/11640), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/4031), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/4062), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/8309), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/8374), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/4131), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/4079), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/6979)
Approved by: Manishearth
Pushing 9a900ef019cd643bff961d7b20db6da69f3edb29 to master...
